db/migrate/20171001035221_move_setting_to_tags.rb

```diff
@@ -0,0 +1,23 @@
+class MoveSettingToTags < ActiveRecord::Migration[5.0]
+  def up
+    print "\n"
+    Character.where.not(setting: nil).each do |character|
+      setting = Setting.where(name: character.setting).first
+      setting ||= Setting.create!(user: character.user, name: character.setting)
+      character.settings << setting
+      print "#{character.name} migrated\n"
+    end
+    print "\n"
+    remove_column :characters, :setting
+  end
+  
+  def down
+    add_column :characters, :setting, :text
+    Setting.all.each do |setting|
```

Comment:
again `find_each`

Reply:
Even after 100% migrating we won't have enough Settings to exceed a single batch, but I'll use it above (which will take 3 batches)